client/src/nv_ingest_client/message_clients/rest/rest_client.py

```diff
@@ -0,0 +1,219 @@
+    ):
+        self._host = host
+        self._port = port
+        self._max_retries = max_retries
+        self._max_backoff = max_backoff
+        self._connection_timeout = connection_timeout
+        self._http_allocator = http_allocator
+        self._client = self._http_allocator()
+        self._retries = 0
+
+        self._submit_endpoint = "/v1/submit_job"
+        self._fetch_endpoint = "/v1/fetch_job"
+
+    def _connect(self) -> None:
+        """
+        Attempts to reconnect to the HTTP server if the current connection is not responsive.
+        """
+        if not self.ping():
+            logger.debug("Reconnecting to HTTP server")
+            self._client = self._http_allocator()
+
+    @property
+    def max_retries(self) -> int:
+        return self._max_retries
+
+    @max_retries.setter
+    def max_retries(self, value: int) -> None:
+        self._max_retries = value
+
+    def get_client(self) -> Any:
+        """
+        Returns a HTTP client instance, reconnecting if necessary.
+
+        Returns
+        -------
+        Any
+            The HTTP client instance.
+        """
+        if self._client is None or not self.ping():
+            self._connect()
+        return self._client
+
+    def ping(self) -> bool:
+        """
+        Checks if the Redis server is responsive.
+
+        Returns
+        -------
+        bool
+            True if the server responds to a ping, False otherwise.
+        """
+        try:
+            self._client.ping()
+            return True
+        except (httpx.HTTPError, AttributeError):
+            return False
+
+    def fetch_message(self, job_state: JobState, timeout: float = 10) -> Optional[str]:
+        """
+        Fetches a message from the specified queue with retries on failure.
+
+        Parameters
+        ----------
+        job_state : JobState
+            The JobState of the message to be fetched.
+        timeout : float
+            The timeout in seconds for blocking until a message is available.
+
+        Returns
+        -------
+        Optional[str]
+            The fetched message, or None if no message could be fetched.
+
+        Raises
+        ------
+        ValueError
+            If fetching the message fails after the specified number of retries or due to other critical errors.
+        """
+        retries = 0
+        while True:
+            try:
+                # Fetch via HTTP
+                url = f"http://{self._host}:{self._port}{self._fetch_endpoint}/{job_state.job_spec.job_id}"
+                result = requests.get(url)
+                logger.debug(f"Fetch Message submitted to http endpoint {self._submit_endpoint}")
+                
+                # If the result contains a 200 then return the raw JSON string response
+                if result.status_code == 200:
+                    return result.text
+                else:
+                    # Follow the established backoff approach
```

Comment:
It would be much better to attempt to make a general function for logging and setting up retries after failures. This was quite lazy on my part

Reply:
Went ahead and added a function `perform_retry_backoff` to handle this